components/camel-aws-sns/src/main/java/org/apache/camel/component/aws/sns/SnsProducer.java

```diff
@@ -104,6 +106,13 @@ private String determineMessageStructure(Exchange exchange) {
                     mav.setDataType("String");
                     mav.withStringValue(value.toString());
                     result.put(entry.getKey(), mav);
+                } else if (value instanceof List && ((List) value).size()>0) {
+                    // Avoiding reliance on .toString()
+                    String delimiter = "\", \"", prefix = "[\"", suffix = "\"]";
+                    MessageAttributeValue mav = new MessageAttributeValue();
+                    mav.setDataType("String.Array");
+                    mav.withStringValue((String) ((List)value).stream().map(Object::toString).collect(Collectors.joining(delimiter, prefix, suffix)));
```

Comment:
If you cast `value` to `List<?>` the compiler warning will disapear and you will not need that cast to String

Reply:
+1

Reply:
Done